src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -15,32 +15,30 @@ namespace System.IO
     //
     public class BinaryWriter : IDisposable, IAsyncDisposable
     {
+        private const int MaxArrayPoolRentalSize = 1024 * 1024; // ArrayPool<T>.Shared allocates beyond this point
```

Comment:
it would have been great if `ArrayPool` was exposing the value as an `internal const`.